x-pack/plugins/cloud_security_posture/public/pages/vulnerabilities/vulnerabilities.tsx

```diff
@@ -4,17 +4,236 @@
+  filters,
+  sort: [{ id: 'cvss', direction: 'desc' }],
+  pageIndex: 0,
+});
 
 export const Vulnerabilities = () => {
+  const { data, isLoading, error } = useFilteredDataView(LATEST_VULNERABILITIES_INDEX_PATTERN);
+
+  if (error) {
+    return <ErrorCallout error={error as Error} />;
+  }
+  if (isLoading || !data) {
+    return <EuiLoadingSpinner />;
+  }
+
+  return <VulnerabilitiesContent dataView={data} />;
+};
+
+const VulnerabilitiesContent = ({ dataView }: { dataView: DataView }) => {
+  const {
+    pageIndex,
+    query,
+    sort,
+    queryError,
+    pageSize,
+    onChangeItemsPerPage,
+    onChangePage,
+    onSort,
+    setUrlQuery,
+  } = useCloudPostureTable({
+    dataView,
+    defaultQuery: getDefaultQuery,
+    paginationLocalStorageKey: LOCAL_STORAGE_PAGE_SIZE_FINDINGS_KEY,
+  });
+  const { euiTheme } = useEuiTheme();
+
+  const { data, isLoading } = useLatestVulnerabilities({
+    query,
+    sort,
+    enabled: !queryError,
+  });
+
+  const renderCellValue = useMemo(() => {
+    return ({
+      rowIndex,
+      columnId,
+    }: {
+      rowIndex: number;
+      columnId: typeof columns[number]['id'];
+    }) => {
+      const vulnerability = data?.page[rowIndex] as VulnerabilityRecord;
+
+      if (!vulnerability) return null;
+
+      if (columnId === 'actions') {
+        return (
+          <EuiButtonIcon
+            iconType="expand"
+            aria-label="View"
+            onClick={() => {
+              alert(`Flyout id ${vulnerability.finding?.vulnerability?.id}`);
+            }}
+          />
+        );
+      }
+      if (columnId === 'vulnerability') {
+        return vulnerability.finding?.vulnerability?.id || null;
+      }
+      if (columnId === 'cvss') {
+        if (!vulnerability.finding?.vulnerability.score.base) {
+          return null;
+        }
+        return (
+          <CVSScoreBadge
+            score={vulnerability.finding.vulnerability.score.base}
+            version={vulnerability.finding?.vulnerability.score.version}
+          />
+        );
+      }
+      if (columnId === 'resource') {
+        return vulnerability.resource?.name || null;
+      }
+      if (columnId === 'severity') {
+        if (
+          !vulnerability.finding?.vulnerability.score.base ||
+          !vulnerability?.finding.vulnerability.severity
+        ) {
+          return null;
+        }
+        return (
+          <SeverityStatusBadge
+            score={vulnerability.finding.vulnerability.score.base}
+            status={vulnerability.finding.vulnerability.severity}
+          />
+        );
+      }
+      if (columnId === 'package-version') {
+        return (
+          <>
+            {vulnerability.finding?.vulnerability.package.name}{' '}
+            {vulnerability.finding?.vulnerability.package.version}
+          </>
+        );
+      }
+      if (columnId === 'fix-version') {
+        return (
+          <>
+            {vulnerability.finding?.vulnerability.package.name}{' '}
```

Comment:
If `vulnerability. finding? can be undefined, then  I believe  `vulnerability`, `package`, and `name` can also be undefined. We are missing ternary operators

Reply:
good catch